engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -0,0 +1,207 @@
+        'Content-Type' => 'application/json',
+        'Authorization' => auth
+      }
+    end
+
+    def prepare_scc_announce_request(uri_path, auth, params)
+      scc_request = Net::HTTP::Post.new(uri_path, headers(auth))
+      hw_info_keys = %i[cpus sockets hypervisor arch uuid cloud_provider]
+      hw_info = params['hwinfo'].symbolize_keys.slice(*hw_info_keys)
+      scc_request.body = {
+        hostname: params['hostname'],
+        hwinfo: hw_info
+      }.to_json
+      scc_request
+    end
+
+    def prepare_scc_request(uri_path, product, auth, token, email)
+      scc_request = Net::HTTP::Post.new(uri_path, headers(auth))
+      scc_request.body = {
+        token: token,
+        identifier: product.identifier,
+        version: product.version,
+        arch: product.arch,
+        release_type: product.release_type,
+        email: email || nil
+      }.to_json
+      scc_request
+    end
+
+    def announce_system_scc(auth, params)
+      uri = URI.parse(ANNOUNCE_URL)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      scc_request = prepare_scc_announce_request(uri.path, auth, params)
+      response = http.request(scc_request)
+      response.error! unless response.code_type == Net::HTTPCreated
+
+      JSON.parse(response.body)
+    end
+
+    def scc_activate_product(product, auth, token, email)
+      uri = URI.parse(ACTIVATE_PRODUCT_URL)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      scc_request = prepare_scc_request(uri.path, product, auth, token, email)
+      http.request(scc_request)
+    end
+
+    def parse_error(error_message, token = nil, email = nil)
+      error_message = error_message[0..(error_message.index('json') - 1)].strip
+      error_message = error_message.gsub(token, '').squish if token
+      error_message = error_message.gsub(email, '').strip if email
+      error_message
+    end
+
+    def scc_check_subscription_expiration(headers, login, logger)
+      auth = headers['HTTP_AUTHORIZATION'] if headers.include?('HTTP_AUTHORIZATION')
+      uri = URI.parse(SYSTEMS_ACTIVATIONS_URL)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      scc_request = Net::HTTP::Get.new(uri.path, headers(auth))
+      response = http.request(scc_request)
+      unless response.code_type == Net::HTTPOK
+        logger.info "Could not get the system (#{login}) activations, error: #{response.message} #{response.code}"
+        response.message = SccProxy.parse_error(response.message) if response.message.include? 'json'
+        return { is_active: false, message: response.message }
+      end
+      scc_systems_activations = JSON.parse(response.body)
+      return { is_active: false, message: 'No activations.' } if scc_systems_activations.empty?
+
+      no_status_products_ids = scc_systems_activations.map {|act| act['service']['product']['id'] if (act['status'].nil? && act['expires_at'].nil?)}.flatten.compact
```

Comment:
Could you provide me some examples where you encountered this? That seems to be an issue in SCC data / API  if that happens.